server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -4060,6 +4063,38 @@ public void afterRefresh(boolean didRefresh) {
         }
     }
 
+    /**
+     * Returns the shard-level field stats, which includes the number of segments in the latest NRT reader of this shard
+     * and the total number of fields across those segments.
+     */
+    public ShardFieldStats getShardFieldStats() {
+        return shardFieldStats;
+    }
+
+    private class RefreshShardFieldStatsListener implements ReferenceManager.RefreshListener {
+        @Override
+        public void beforeRefresh() {
+
+        }
+
+        @Override
+        public void afterRefresh(boolean didRefresh) {
+            if (shardFieldStats == null || didRefresh) {
+                try (var searcher = getEngine().acquireSearcher("shard_field_stats", Engine.SearcherScope.INTERNAL, Function.identity())) {
```

Comment:
nit: there seem to be a `acquireSearcher(String source, SearcherScope scope)` that could be used here.

Reply:
Yes, fixed in https://github.com/elastic/elasticsearch/pull/111525/commits/0ee2b70bc4424da3938bb8f0c55e4b7ba3589912.